FirebaseAdmin/FirebaseAdmin.Tests/Messaging/MulticastMessageTest.cs

```diff
@@ -0,0 +1,1542 @@
+﻿// Copyright 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FirebaseAdmin.Messaging;
+using Google.Apis.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Action = FirebaseAdmin.Messaging.Action;
+
+namespace FirebaseAdmin.Tests.Messaging
+{
+    public class MulticastMessageTest
+    {
+        [Fact]
+        public void EmptyMulticastMessage()
+        {
+            var message = new MulticastMessage
+            {
+                Tokens = new[] { "test-token" },
+            };
+            this.AssertJsonEquals(new JObject() { { "tokens", new JArray("test-token") } }, message);
+        }
+
+        [Fact]
+        public void GetMessageList()
+        {
+            var message = new MulticastMessage
+            {
+                Tokens = new[] { "test-token1", "test-token2" },
+            };
+
+            var messages = message.GetMessageList();
+
+            Assert.Equal(2, messages.Count);
+            Assert.Equal("test-token1", messages[0].Token);
+            Assert.Equal("test-token2", messages[1].Token);
+        }
+
+        [Fact]
+        public void Data()
+        {
+            var message = new MulticastMessage
+            {
+                Tokens = new[] { "test-token" },
+                Data = new Dictionary<string, string>()
+                {
+                    { "k1", "v1" },
+                    { "k2", "v2" },
+                },
+            };
+            this.AssertJsonEquals(
```

Comment:
Same comment as above. Unit test GetMessageList and make sure we correctly create 2 copies of the message. Applies to most test cases in this class.